### PR TITLE
feat: improved DX and support for running component and e2e tests w/ gulp

### PR DESCRIPTION
**Pull request by JessicaSachs**
- Changes default GraphQL port from `52159` to `52220`
- Improved process cleanup from gulp / exit
- Records ct & e2e tests for new UI to dashboard

### How to test

1. `yarn gulp cyOpenLaunchpadE2E`
2. `yarn gulp cyRunLaunchpadE2E`
3. `yarn gulp cyOpenAppE2E`
4. `yarn gulp cyRunLaunchpadE2E`

### CODE REVIEW

File: packages/frontend-shared/env.d.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Unsure why the ImportMeta type isn't picking this up.
+interface ImportMetaEnv {
+  VITE_CYPRESS_INTERNAL_GQL_PORT: number
+}
```
Comment (JessicaSachs) [inline packages/frontend-shared/env.d.ts:4]: This requires typescript help

File: .vscode/settings.json [hunk 2]
```diff
@@ -21,5 +21,16 @@
   "editor.codeActionsOnSave": {
     "source.fixAll.eslint": true
   },
-  "typescript.tsdk": "node_modules/typescript/lib"
+  "typescript.tsdk": "node_modules/typescript/lib",
+  "cSpell.words": [
```
Comment (tgriesser) [inline .vscode/settings.json:25]: 🤔

Comment (JessicaSachs) [reply .vscode/settings.json:25]: yeah. didnt know how I flt about it. Will switch to User settings I think.

**Comment by wickeddude-star** [comment]
Winks

**Comment by brian-mann** [comment]
Task linked: [PM-13 Run tests in Safari](https://app.clickup.com/t/18033298/PM-13)

**Comment by brian-mann** [comment]
Task linked: [PM-17 🍀 Sessions](https://app.clickup.com/t/18033298/PM-17)

**Comment by lmiller1990** [comment]
I am not sure if this is ready for testing yet, but I tried it out since it's in review. 

`yarn gulp cypressOpenLaunchpad` gave me this error. 
 
Screen Shot 2021-09-20 at 5 05 53 pm

Electron still opens, but it's stuck on 'loading'.

`yarn gulp cypressRunLaunchpad` is running the scaffolded e2e tests inside of `packages/launchpad/cypress/integration`  (as opposed to the component tests, which is what I think it's supposed to be doing).

**Comment by JessicaSachs** [comment]
It's not ready for testing :-). The issue you're getting is an issue solved by changing the NODE_ENV to staging inside the gulp task... which is something fixed that isn't pushed yet, since @tgriesser is going to sync this PR up with how services executes the Cy binary.

**Review by tgriesser — APPROVED** [review]

**Pull request merged**